catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -336,6 +347,35 @@ public void addFields(Metacard metacard, SolrInputDocument solrInputDocument)
         solrInputDocument.addField(SchemaFields.METACARD_TYPE_OBJECT_FIELD_NAME, metacardTypeBytes);
     }
 
+    public static String truncateAsUTF8(String value, int maximumBytes) {
```

Comment:
❓ @spearskw Can you go into a little more detail on what this method is doing and what it means? Some comments in the code might not hurt. 

I presume we're doing manual byte counting because the actual size of bytes per character can vary in UTF-8 Encoding,  Just not sure on the specifics of reasons why the `value.length()` doesn't work and such.

Reply:
I'll be honest, I don't know what it's doing. @pklinef ?